### DISCUSSION TITLE
data: data engineering+ data analyst + data scientist

### DISCUSSION THREAD

**Discussion by TinyHust** [TinyHust/notes#436] [DE]
- [Data Engineering](https://github.com/TinyHust/notes/discussions/436#discussioncomment-11117139)
- [Data Scientist](https://github.com/TinyHust/notes/discussions/436#discussioncomment-11117155)
- [Data Analyst](https://github.com/TinyHust/notes/discussions/436#discussioncomment-11117153)

**Comment by TinyHust** [comment]
## Data Engineering
- [Apache Hadoop](https://hadoop.apache.org/)
- [Apache Spark](https://spark.apache.org/)
- [Apache Kafka](https://kafka.apache.org/)
- [Apache Airflow](https://airflow.apache.org/)
- [Apache Flink](https://flink.apache.org/)
- [Delta Lake](https://delta.io/)
- [Decodable](https://www.decodable.co/)
- [Apache Druid](https://druid.apache.org/)
- [Databricks Community Edition](https://community.cloud.databricks.com/)
- [snowflake](https://github.com/TinyHust/notes/discussions/457)
- [iceberg](https://iceberg.apache.org/)
- [Apache Hudi](https://hudi.apache.org/)
- [OpenLineage](https://openlineage.io/)
- [marquez](https://github.com/MarquezProject/marquez)
- [OpenMetadata](https://open-metadata.org/)
- ETL (Extract, Transform, Load)
  - [AWS Glue](https://aws.amazon.com/glue/)
  - [Google Cloud Dataflow](https://cloud.google.com/dataflow)
  - [Azure Data Factory](https://azure.microsoft.com/en-us/products/data-factory)
    - [Synapse Analytics](https://learn.microsoft.com/en-us/azure/synapse-analytics/)
  - [airbyte](https://airbyte.com/)
  - [databricks](https://www.databricks.com/)
  - [dagster](https://dagster.io/)
  - [DBT](https://www.getdbt.com/) (Data Build Tool)
  - [dlthub](https://dlthub.com/)
- CDC (Change Data Capture)
  - [Debezium](https://debezium.io/)
- Visualizations
  - [tableau](https://www.tableau.com/)
  - [power-bi](https://app.powerbi.com/)
  - [superset](https://superset.apache.org/)
  - [streamlit](https://github.com/streamlit/streamlit)
  - [pygwalker](https://github.com/Kanaries/pygwalker)
  - [metabase](https://github.com/metabase/metabase)
  - [cube](https://github.com/cube-js/cube)
- [columnar data](https://en.wikipedia.org/wiki/Column-oriented_DBMS)
  - BigQuery
  - Redshift
- Unit Tests
  - [deequ](https://github.com/awslabs/deequ)

**Reply by mediumhust** [reply]
### DataOps

---
- [Stream data ingestion with Redpanda](https://medium.com/@ongxuanhong/dataops-01-stream-data-ingestion-with-redpanda-56b7fd768887)
  - stack: Redpanda, MinIo, Debezium
  - https://github.com/mediumhust/de01-stream-ingestion-redpanda-minio
- [Spawn up Apache Spark infrastructure by using Docker](https://medium.com/@ongxuanhong/dataops-02-spawn-up-apache-spark-infrastructure-by-using-docker-fec518698993)
  - https://github.com/mediumhust/de02-pyspark-optimization
- [Trino + DBT + Spark](https://medium.com/@ongxuanhong/dataops-03-trino-dbt-spark-everything-everywhere-all-at-once-241932d27a6)
  - https://github.com/mediumhust/de03-trino-dbt-spark-everything-everywhere-all-at-once
  - https://www.kaggle.com/datasets/olistbr/brazilian-ecommerce
- [Iceberg + Spark + Trino + Dagster](https://blog.devgenius.io/modern-data-stack-demo-5d75dcdfba50)
  - https://github.com/mediumhust/ngods-stocks
- [Nessie, MinIO, Spark, and Dremio](https://dev.to/alexmercedcoder/hands-on-with-apache-iceberg-on-your-laptop-deep-dive-with-apache-spark-nessie-minio-dremio-polars-and-seaborn-2hgk)
  - https://github.com/mediumhust/dremio-spark-nessie-demo-environment-template
- [Stackable](https://github.com/stackabletech)
  - Apache Airflow
  - Apache Druid
  - Apache HBase
  - Apache Hadoop HDFS
  - Apache Hive Metastore
  - Apache Kafka
  - Apache NiFi
  - Apache Spark
  - Apache Superset
  - Trino
  - Apache ZooKeeper
- AWS stack: Airflow, S3, Glue, Redshift
  - https://github.com/mediumhust/aws-airflow-dataengineering-pipeline
  - https://www.youtube.com/watch?v=XtDbn9-pGh0

**Reply by mediumhust** [reply]
Samples:
- [dlt](https://github.com/mediumhust/pythondataanalysis/tree/main/ETL%20Pipeline/dltproject)
- dagster:
  - [io manager](https://github.com/hnawaz007/pythondataanalysis/tree/main/dagster-project/etl)
  - https://github.com/mediumhust/modern-data-stack-etl-

**Comment by TinyHust** [comment]
## Data Scientist
- [Cookiecutter](https://cookiecutter-data-science.drivendata.org/)
- [mlflow](https://mlflow.org/)

**Reply by TinyHust** [reply]
**REFs:**

**Comment by TinyHust** [comment]
## Data Analyst
